nutanix/resource_nutanix_karbon_cluster_worker_pool.go

```diff
@@ -0,0 +1,400 @@
+package nutanix
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-nutanix/client/karbon"
+	"github.com/terraform-providers/terraform-provider-nutanix/utils"
+)
+
+func resourceNutanixKarbonWorkerNodePool() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNutanixKarbonWorkerNodePoolCreate,
+		ReadContext:   resourceNutanixKarbonWorkerNodePoolRead,
+		UpdateContext: resourceNutanixKarbonWorkerNodePoolUpdate,
+		DeleteContext: resourceNutanixKarbonWorkerNodePoolDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(DEFAULTWAITTIMEOUT * time.Minute),
+			Update: schema.DefaultTimeout(DEFAULTWAITTIMEOUT * time.Minute),
+			Delete: schema.DefaultTimeout(DEFAULTWAITTIMEOUT * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"cluster_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"node_os_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"num_instances": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     false,
+				ValidateFunc: validation.IntAtLeast(MINNUMINSTANCES),
```

Comment:
Where is this constant declared?

Reply:
This constant is declared in cluster resource. and K8s requires minimum config to add worker nodepool